addons/controllers/clusterbootstrap_controller.go

```diff
@@ -1096,25 +1095,37 @@ func (r *ClusterBootstrapReconciler) watchProvider(providerRef *corev1.TypedLoca
 // - string: The secret name which references to the Secret CR on mgmt cluster under a particular cluster namespace.
 // - error: whether there is error when getting the secret name.
 func (r *ClusterBootstrapReconciler) GetDataValueSecretNameFromBootstrapPackage(cbPkg *runtanzuv1alpha3.ClusterBootstrapPackage, cluster *clusterapiv1beta1.Cluster) (string, error) {
-	// When valuesFrom is nil, we interpret it as no data values are needed for the package installation.
-	if cbPkg.ValuesFrom == nil {
-		r.Log.Info(fmt.Sprintf("no data values are provided to the ClusterBootstrapPackage.ValuesFrom field. ClusterBootstrapPackage.RefName: %s", cbPkg.RefName))
-		return "", nil
-	}
+	var (
+		packageRefName string
+		err            error
+	)
 
-	if cbPkg.ValuesFrom.Inline != nil {
-		packageRefName, _, err := util.GetPackageMetadata(r.context, r.aggregatedAPIResourcesClient, cbPkg.RefName, cluster.Namespace)
+	if cbPkg.ValuesFrom == nil || cbPkg.ValuesFrom.Inline != nil {
+		packageRefName, _, err = util.GetPackageMetadata(r.context, r.aggregatedAPIResourcesClient, cbPkg.RefName, cluster.Namespace)
 		if packageRefName == "" || err != nil {
 			// Package.Spec.RefName and Package.Spec.Version are required fields for Package CR. We do not expect them to be
 			// empty and error should not happen when fetching them from a Package CR.
 			r.Log.Error(err, fmt.Sprintf("unable to fetch Package.Spec.RefName or Package.Spec.Version from Package %s/%s",
 				cluster.Namespace, cbPkg.RefName))
 			return "", err
 		}
+	}
+
+	if cbPkg.ValuesFrom == nil {
```

Comment:
Question: Do you think if we refactored this to something like below it would be more readable?
```
if cbPkg.ValuesFrom != nil {
 // existing logic that handles inline, secretRef and ProviderRef
} else {
// create empty secret for TKGS
}
```